lib/commands/ime.js

```diff
@@ -0,0 +1,48 @@
+import log from '../logger';
+import { errors } from 'mobile-json-wire-protocol';
+
+let commands = {}, helpers = {}, extensions = {};
+
+commands.isIMEActivated = async function () {
+  // IME is always activated on Android devices
+  return true;
+};
+
+commands.availableIMEEngines = async function () {
+  log.debug('Retrieving available IMEs');
+  let engines = await this.adb.availableIMEs();
+  log.debug(`Engines: ${JSON.stringify(engines)}`);
+  return engines;
+};
+
+commands.getActiveIMEEngine = async function () {
+  log.debug('Retrieving current default IME');
+  let engine = await this.adb.defaultIME();
+  return engine;
+};
+
+commands.activateIMEEngine = async function (imeId) {
+  log.debug(`Attempting to activate IME ${imeId}`);
+  let availableEngines = await this.adb.availableIMEs();
+  if (availableEngines.indexOf(imeId) !== -1) {
```

Comment:
This case should be reversed, so the error case is first. Then the rest can be outside of the `if`.